### PR TITLE
feat: not connecting to its own addresses

### PR DESCRIPTION
**Pull request by michael1011**
Closes https://github.com/ExchangeUnion/xud/issues/1091

After thinking a little bit about it I am of the opinion that only checking whether the address belongs to itself when trying to connect is sufficient. All other checks could cause weird edge cases and inconsistent behaviors and that single check is enough to fix the issue.

### CODE REVIEW

File: test/jest/Pool.spec.ts [hunk 1]
```diff
@@ -66,4 +71,20 @@ describe('P2P Pool', () => {
     expect(pool.getTokenIdentifier('DAI')).toEqual(daiTokenAddress);
     expect(pool['nodeState'].raidenAddress).toEqual(raidenAddress);
   });
+
+  test('should reject connecting to its own addresses', async () => {
```
Comment (erkarl) [inline test/jest/Pool.spec.ts]: :)

**Comment by erkarl** [comment]
Working as advertised...
